metadata-utils/src/main/java/com/linkedin/metadata/utils/IngestionUtils.java

```diff
@@ -0,0 +1,35 @@
+package com.linkedin.metadata.utils;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+
+public class IngestionUtils {
+
+  private static final String PIPELINE_NAME = "pipeline_name";
+
+  private IngestionUtils() {
+  }
+
+  /**
+   * Injects a pipeline_name into a recipe if there isn't a pipeline_name already there.
+   * The pipeline_name will be the urn of the ingestion source.
+   *
+   * @param pipelineName the new pipeline name in the recipe.
+   * @return a modified recipe JSON string
+   */
+  public static String injectPipelineName(final String originalJson, final String pipelineName) {
+    try {
+      JSONObject jsonRecipe = new JSONObject(originalJson);
+      boolean hasPipelineName = jsonRecipe.has(PIPELINE_NAME) && jsonRecipe.get(PIPELINE_NAME) != null && !jsonRecipe.get(PIPELINE_NAME).equals("");
+
+      if (!hasPipelineName) {
+        jsonRecipe.put(PIPELINE_NAME, pipelineName);
+        return jsonRecipe.toString();
+      }
+    } catch (JSONException e) {
```

Comment:
We can also bubble up the underlying exception so we can see what was wrong with the JSON:

```
throw new IllegalArgumentException("Failed to create execution request: Invalid recipe json provided.", e);
```

Reply:
nice definitely